plugins/imgur.py

```diff
@@ -128,7 +129,15 @@ def export_submission(self,
             if is_album:
                 results['link_display'] = '[Imgur Album](https://imgur.com/a/%s)  \n' % album['id']
             else:
-                results['link_display'] = '[Imgur](%s)  \n' % images[0]['link'].replace('http', 'https')
+                
+                picture_url = images[0]['link'].replace('http', 'https')
+                r = requests.head(picture_url)
+                mime_text = r.headers.get('Content-Type')
+                mime = mimeparse.parse_mime_type(mime_text)
+                if mime[1] == 'gif':
+                    picture_url = re.sub('\.gif|jpg|png', '.gifv', picture_url)
```

Comment:
Sorry, Python isn't my first language. I'm not very used to it. So I didn't know that `\.` was deprecated. I'm not quite sure what you mean by a `rsplit`, what would I be splitting by? 

Does this example make any sense?
```
pictureurl = 'https://i.imgur.com/iGAq9SV.gif'
split = pictureurl.rsplit('.')
split[3] = 'gifv'
pictureurl = '.'.join(split)
```

This would forgo the use of regex entirely, anyway.